test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +111,127 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx)(error) {
+// cteate namespace
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := ctx.Client.CoreV1().Namespaces().Get(context.Background(), ctx.Namespace, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+	return nil
+}
+	if apierrors.IsNotFound(err) {
+		n := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ctx.Namespace,
+			Labels: map[string]string{
+				"name": ctx.Namespace + "-" + testingNamespaceLabel,
+			},
+		},
+	}
+	_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), n, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+}
+
+//decode and create yamls except for persistentvolume
+
+type nfsMainifestItem struct {
+	resourceFile strings
+	action [2]string
+}
+
+var nfsManifestFiles = []nfsMainifestItem {
+	nfsServer {
+		resourceFile: "/test/testhelpers/nfs_server.yaml",
+		action: [2]string{"CREATE"},
+	},
+	nfsService {
+		resourceFile: "/test/testhelpers/nfs_service.yaml",
+		action: [2]string{"CREATE", "GET"},
+	},
+	nfsStorageclass {
+		resourceFile: "/test/testhelpers/nfs_storageclass.yaml",
+		action: [2]string{"CREATE"}
+	}
+}
+
+
+for _, manifests := range nfsManifestFiles {
+// err = deploy.DecodeYAMLManifestToObject(filepath.join(repoDir, nfsManifestsDir, nfsServerManifestFilename), ...)
+
+	err := deploy.DecodeYAMLManifestToObject(manifest, &nfs_server, &nfs_service, &nfs_storageclass)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepath, err)
+	}
+		if nfsManifestItem.action == "CREATE" {
+			ctx.Client.CoreV1().Namespaces().Create(context.Background(nfs_server,nfs_service, nfs_storageclass), &nfs_server, &nfs_service, &nfs_storageclass, metav1.CreateOptions{})
+	}
+			if err != nil {
+				return err
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfs_server, nfs_service, nfs_storageclass)
+	} //
+	 	if nfsManifestItem.action == GET {
+			_, err = ctx.Client.CoreV1().Service(ctx.Namespace).Get(context.Background(), "nfs-service", metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return nil
+			}
+
```

Comment:
Hmm, this could be problematic, especially in the case where the resource already exists. It's better to try and get the resource first, check if it already exists, and if it doesn't, create it.